vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1012,6 +1027,23 @@ func resourceVSphereVirtualMachineCustomizeDiff(_ context.Context, d *schema.Res
 		return err
 	}
 
+        // Check the power state diff for virtual machine
+	power_state := d.Get("power_state").(string)
+	switch power_state {
+	case "poweredOn":
```

Comment:
This logic is very confusing to me... but more importantly I don't believe there needs to be any customize diffing around computed attributes. This can be removed.

Reply:
I'll take a look into it, Alex and report back.

Reply:
I've removed this and have retested. ✅